### ISSUE TITLE
Missing UpdateAsync(...) overload in ElasticsearchClient

### ISSUE THREAD

**Issue by mynkow** [elastic/elasticsearch-net#8120]
**Elastic.Clients.Elasticsearch version**: 8.13.5

**Elasticsearch version**: 8.5.1

**.NET runtime version**: 8.0.200

**Operating system version**: win11

**Description of the problem including expected versus actual behavior**:
After updating from v 8.12.1 to 8.13.5 the following method does not exist anymore in `ElasticsearchClient`. I see that the class is autogenerated and I am not sure if this is on purpose or there is an issue in code generation:

```
	/// <summary>
	/// <para>Updates a document with a script or partial document.</para>
	/// <para><see href="https://www.elastic.co/guide/en/elasticsearch/reference/master/docs-update.html">Learn more about this API in the Elasticsearch documentation.</see></para>
	/// </summary>
	public virtual Task<UpdateResponse<TDocument>> UpdateAsync<TDocument, TPartialDocument>(TDocument document, TPartialDocument partialDocument, Action<UpdateRequestDescriptor<TDocument, TPartialDocument>> configureRequest, CancellationToken cancellationToken = default)
	{
		var descriptor = new UpdateRequestDescriptor<TDocument, TPartialDocument>(document);
		configureRequest?.Invoke(descriptor);
		descriptor.BeforeRequest();
		return DoRequestAsync<UpdateRequestDescriptor<TDocument, TPartialDocument>, UpdateResponse<TDocument>, UpdateRequestParameters>(descriptor, cancellationToken);
	}
```

Can you please confirm what is going on?

**Comment by flobernd** [comment]
Hi @mynkow,

thanks for reporting. This seems to be an issue with the code generator. `index` and `id` are required values for the `UpdateRequest` which causes the function to generate like this right now:

```csharp
public virtual Task<UpdateResponse<TDocument>> UpdateAsync<TDocument, TPartialDocument>(Elastic.Clients.Elasticsearch.IndexName index, Elastic.Clients.Elasticsearch.Id id, Action<UpdateRequestDescriptor<TDocument, TPartialDocument>> configureRequest, CancellationToken cancellationToken = default)
```

However, as `index` and `id` can be infered, we should probably add back the method without the required parameters. Same applies to other places, like `CreateAsync`, etc. it seems from a brief check.

**Comment by flobernd** [comment]
I adjusted the code generator to emit an exhaustive set of overloads for all request methods. This includes generic and non-generic versions as well as all variants of the `document`, `index`, `id` overloads and variants with- and without a descriptor action parameter.

**Comment by flobernd** [comment]
Fixed in 8.13.6
